File: Fresh/Fresh.Services/Database/ProductPrice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Fresh.Services.Database
{
    public class ProductPrice
    {
        [Key]
        public int Id { get; set; }
        public decimal PricePerUnit { get; set; }
        public DateTime EffectiveFrom { get; set; }

        [ForeignKey(nameof(ProductId))]
        public Products Products { get; set; }
        public int ProductId { get; set; }
    }
}

File: Fresh/UI/fresh_app/src/app/model/ProductPrice.ts
import { Products } from "./Products";

export class ProductPrice {
    id: number;
    pricePerUnit: number;
    effectiveFrom: Date;
    productId: number;

    products: Products;

    constructor(id: number, pricePerUnit: number, effectiveFrom: Date, productId: number, products: Products) {
        this.id = id;
        this.pricePerUnit = pricePerUnit;
        this.effectiveFrom = effectiveFrom;
        this.productId = productId;
        this.products = products;
    }
}